packages/commonwealth/server/routes/spam/markCommentAsSpam.ts

```diff
@@ -0,0 +1,69 @@
+import { AppError } from 'common-common/src/errors';
+import type { NextFunction, Request, Response } from 'express';
+import type { DB } from '../../models';
+import { Op, Sequelize } from 'sequelize';
+import { success } from '../../types';
+import { findAllRoles } from '../../util/roles';
+
+export const Errors = {
+  InvalidCommentId: 'Comment ID invalid',
+  NotLoggedIn: 'Not logged in',
+  CommentNotFound: 'Could not find Comment',
+  NotAdmin: 'Not an admin',
+};
+
+export default async (
+  models: DB,
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const commentId = req.params.id;
+  if (!commentId) {
+    return next(new AppError(Errors.InvalidCommentId));
+  }
+
+  if (!req.user) {
+    return next(new AppError(Errors.NotLoggedIn));
+  }
+
+  const comment = await models.Comment.findOne({
+    where: {
+      id: commentId,
+    },
+  });
+  if (!comment) {
+    return next(new AppError(Errors.CommentNotFound));
+  }
+  const userOwnedAddressIds = (await req.user.getAddresses())
+    .filter((addr) => !!addr.verified)
+    .map((addr) => addr.id);
+  if (!userOwnedAddressIds.includes(comment.address_id)) {
+    // is not author
+    const roles = await findAllRoles(
+      models,
+      { where: { address_id: { [Op.in]: userOwnedAddressIds } } },
+      comment.chain,
+      ['admin', 'moderator']
+    );
+    const role = roles.find((r) => {
+      return r.chain_id === comment.chain;
+    });
+    if (!role) {
+      return next(new AppError(Errors.NotAdmin));
+    }
+  }
+
+  await comment.update({
```

Comment:
If you wanted to clean this up a little bit, we could instead use `comment.marked_as_spam_at = Date.now(); await comment.save();` and then we wont need to refetch the comment.